src/common/helper/limitExecution/debounce.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Debounces a function call. It returns a new function debouncing
+ * the original one.
+ * Debouncing enforces a function not to be called again until a
+ * certain amount of time has passed without it being called.
+ * For example, "execute this function only if 100 milliseconds
+ * have passed without it being called".
+ */
+function debounce(func, threshold: number = 10): any {
```

Comment:
Why do we need debounce? in which case?

Reply:
We are not using `debouce` sorry, we are using `throttle` instead. Reason: to slow down callback calls when scrolling, I didn't want scroll synchronization feature to make the app less responsive.
I implemented them cause they are tiny functions and I didn't know about `lodash`, good to know.  No problem for me to replace for `lodash.throttle`. Do you prefer it?

Reply:
I did it anyway. Replaced for `lodash.throttle`. [Link to commit](https://github.com/Lemoncode/LeanMood/pull/168/commits/7b6662643273871e599711520a100bc7c89f6632).

**Done**